### PR TITLE
fix: Increase timeout for initial sync from 10 seconds to 2 minutes

### PR DESCRIPTION
**Pull request by krille-chan**
Tested in FluffyChat on iOS and worked good for me. Initial sync was much faster :-)

This fixes the response
timeout for the initial
sync and sets it to 2
minutes instead of 10
seconds. This should increase
the speed for the initial
sync especially for large
accounts. This change
also adds some
documentation in the
code about what
timeout does what.

### CODE REVIEW

File: lib/src/client.dart [hunk 1]
```diff
@@ -1702,8 +1707,15 @@ class Client extends MatrixApi {
...
-          await syncRequest.timeout(timeout + const Duration(seconds: 10));
+
+      // The timeout for the response from the server. If we do not set a sync
+      // timeout (for initial sync) we give the server a longer time to
+      // responde.
+      final responseTimeout = timeout.inMilliseconds == 0
+          ? const Duration(minutes: 2)
+          : timeout + const Duration(seconds: 10);
+
+      final syncResp = await syncRequest.timeout(responseTimeout);
```
Comment (nico-famedly) [inline lib/src/client.dart:1718]: Do we even need a timeout for initial sync? It will fail eventually anyway, if there was a network error or such, right?

Comment (krille-chan) [reply lib/src/client.dart:1718]: No, it is possible that it loads forever with the http package. Happens from time to time

File: lib/src/client.dart [hunk 2]
```diff
@@ -1702,8 +1707,15 @@ class Client extends MatrixApi {
       });
       _currentSyncId = syncRequest.hashCode;
       onSyncStatus.add(SyncStatusUpdate(SyncStatus.waitingForResponse));
-      final syncResp =
-          await syncRequest.timeout(timeout + const Duration(seconds: 10));
+
+      // The timeout for the response from the server. If we do not set a sync
+      // timeout (for initial sync) we give the server a longer time to
+      // responde.
+      final responseTimeout = timeout.inMilliseconds == 0
```
Comment (nico-famedly) [inline lib/src/client.dart:1714]:
```suggestion
      final responseTimeout = timeout == Duration.zero
```

**Review by nico-famedly — APPROVED** [review]

File: lib/src/client.dart (hunk 2 above)
Comment (krille-chan) [reply lib/src/client.dart]: ah thanks. I have searched for something like `Duration.isZero` and was confused that I didn't found this. Now I know why :D

**Pull request merged**